modules/seqexec/server/src/main/scala/seqexec/server/gems/Gems.scala

```diff
@@ -24,25 +33,71 @@ trait Gems[F[_]] extends Gaos[F] {
+                                                             guideConfigDb: GuideConfigDb[F]
+                                                            ) extends Gems[F] {
 
     override val cfg: GemsConfig = config
 
-    override def observe(config: Either[AltairConfig, GemsConfig], expTime: Time): F[Unit] = controller.observe(expTime)
+    override def observe(config: Either[AltairConfig, GemsConfig], expTime: Time): F[Unit] = ().pure[F]
 
-    override def endObserve(config: Either[AltairConfig, GemsConfig]): F[Unit] = controller.endObserve
+    override def endObserve(config: Either[AltairConfig, GemsConfig]): F[Unit] = ().pure[F]
 
-    override def pauseResume(config: GemsConfig,
-                             pauseReasons: PauseConditionSet,
-                             resumeReasons: ResumeConditionSet): F[PauseResume[F]] = {
-      controller.pauseResume(cfg, pauseReasons, resumeReasons)
-    }
+    override def pauseResume(pauseReasons: PauseConditionSet,
+                             resumeReasons: ResumeConditionSet): F[PauseResume[F]] =
+      guideConfigDb.value.map(_.gaosGuide).flatMap{
+        case Some(Right(gemsCfg)) => controller.pauseResume(pauseReasons, resumeReasons)(combine(gemsCfg, cfg))
+        case _                    =>
+          SeqexecFailure.Execution("Attempting to run GeMS sequence before GeMS has being configured.")
+            .raiseError[F, PauseResume[F]]
+      }
 
     override val stateGetter: GemsWfsState[F] = controller.stateGetter
   }
 
-  def fromConfig[F[_]: Applicative](controller: GemsController[F])/*(config: Config)*/: F[Gems[F]] =
-    Applicative[F].pure(new GemsImpl[F](controller, GemsOff))
+  // `combine` calculates the final configuration between the configuration coming from the step and the configuration
+  // set by the operator.
+  private def combine(opConfig: GemsConfig, stepConfig: GemsConfig): GemsConfig = GemsController.GemsOn(
+    Cwfs1Usage.fromBoolean(opConfig.isCwfs1Used && stepConfig.isCwfs1Used),
+    Cwfs2Usage.fromBoolean(opConfig.isCwfs2Used && stepConfig.isCwfs2Used),
+    Cwfs3Usage.fromBoolean(opConfig.isCwfs3Used && stepConfig.isCwfs3Used),
+    Odgw1Usage.fromBoolean(opConfig.isOdgw1Used && stepConfig.isOdgw1Used),
+    Odgw2Usage.fromBoolean(opConfig.isOdgw2Used && stepConfig.isOdgw2Used),
+    Odgw3Usage.fromBoolean(opConfig.isOdgw3Used && stepConfig.isOdgw3Used),
+    Odgw4Usage.fromBoolean(opConfig.isOdgw4Used && stepConfig.isOdgw4Used),
+    P1Usage.fromBoolean(opConfig.isP1Used && stepConfig.isP1Used),
+    OIUsage.fromBoolean(opConfig.isOIUsed && stepConfig.isOIUsed)
+  )
+
+  def fromConfig[F[_]: MonadError[?[_], Throwable]](controller: GemsController[F], guideConfigDb: GuideConfigDb[F])
+                                                         (config: Config)
+  : TrySeq[Gems[F]] = {
+    for {
+      p1    <- config.extractAs[StandardGuideOptions.Value](TELESCOPE_KEY / Tcs.GUIDE_WITH_PWFS1_PROP)
```

Comment:
There are utility methoods `extractObsAs` and `extractInstAs`. Maybe you can add a `extractTelescopeAs`

Reply:
That makes a lot of sense.